contracts/02-tamagotchi-interaction/src/lib.rs

```diff
@@ -24,17 +70,73 @@ extern fn handle() {
     // TODO: 0️⃣ Copy the `handle` function from the previous lesson and push changes to the master branch
     let action: TmgAction = msg::load().expect("unable to load action");
     let tmg = unsafe { TAMAGOTCHI.get_or_insert(Default::default()) };
-    match action {
-        TmgAction::Name => {
-            msg::reply(TmgEvent::Name(tmg.name.clone()), 0)
-                .expect("Error in a reply'tamagotchi::name'");
-        }
-        TmgAction::Age => {
-            let age = exec::block_timestamp() - tmg.date_of_birth;
-            msg::reply(TmgEvent::Age(age), 0).expect("Error in a reply'tamagotchi::age'");
+    if msg::source() == tmg.owner {
+        match action {
+            TmgAction::Name => {
+                msg::reply(TmgEvent::Name(tmg.name.clone()), 0)
+                    .expect("Error in a reply'tamagotchi::name'");
+            }
+            TmgAction::Age => {
+                let age = exec::block_timestamp() - tmg.date_of_birth;
+                msg::reply(TmgEvent::Age(age), 0).expect("Error in a reply'tamagotchi::age'");
+            }
+            TmgAction::Feed => {
+                if tmg.current_fed() <= 9000 {
+                    let fed = tmg.fed + FILL_PER_FEED;
+                    msg::reply(TmgEvent::Fed, 0).expect("Error in a reply'tamagotchi::fed'");
+                    tmg.fed = fed;
+                    tmg.fed_block = exec::block_height() as u64;
+                    tmg.entertained = tmg.current_entertained();
+                    tmg.slept = tmg.current_slept();
+                } else {
+                    let fedblock = exec::block_height() as u64;
+                    tmg.fed = 10000;
+                    tmg.fed_block = fedblock;
+                    tmg.entertained = tmg.current_entertained();
+                    tmg.slept = tmg.current_slept();
```

Comment:
The same code is in both `if` branches.

Reply:
<img width="445" alt="image" src="https://github.com/Jackliu-miaozi/Jackliu-gear-academy/assets/71253778/17152358-72dc-46ed-aa28-637fbee9bd93">